tools/test/examples/examples_lib.py

```diff
@@ -444,3 +476,18 @@ def update_mbedos_version(config, tag, examples):
     
     return 0
   
+def save_test_spec(json_spec, name="test_spec.json"):
+    """save the given json data to test_spec.json"""
+    print ("Dumping json test_specs file {}".format(name))
+    with open(name, 'w') as outfile:  
+        json.dump(json_spec, outfile , indent=4)
```

Comment:
I believe we have an existing function that you can use: https://github.com/ARMmbed/mbed-os/blob/master/tools/utils.py#L184

Reply:
Thanks for the review, all your comments been addressed